scipy/signal/tests/test_signaltools.py

```diff
@@ -2626,79 +2686,102 @@ def test_sine(self, xp):
         n = int(np.ceil(np.log(eps) / np.log(r)))
 
         # High order lowpass filter...
-        y = self.filtfilt(zpk, x, padlen=n)
+        y = self.filtfilt(zpk, x, padlen=n, xp=xp)
         # Result should be just xlow.
         err = np.abs(y - xlow).max()
         assert err < 1e-4
 
         # A 2D case.
-        x2d = np.vstack([xlow, xlow + xhigh])
-        y2d = self.filtfilt(zpk, x2d, padlen=n, axis=1)
+        x2d = xp.asarray(np.vstack([xlow, xlow + xhigh]))
+        y2d = self.filtfilt(zpk, x2d, padlen=n, axis=1, xp=xp)
         assert y2d.shape == x2d.shape
         err = np.abs(y2d - xlow).max()
         assert err < 1e-4
 
         # Use the previous result to check the use of the axis keyword.
         # (Regression test for ticket #1620)
-        y2dt = self.filtfilt(zpk, x2d.T, padlen=n, axis=0)
+        y2dt = self.filtfilt(zpk, x2d.T, padlen=n, axis=0, xp=xp)
         xp_assert_equal(y2d, y2dt.T)
 
+    @skip_xp_backends('torch', reason='negative strides')
     def test_axis(self, xp):
+        if is_jax(xp) and self.filtfilt_kind == 'sos':
+            pytest.skip(reason='sosfilt works in-place')
+
         # Test the 'axis' keyword on a 3D array.
         x = np.arange(10.0 * 11.0 * 12.0).reshape(10, 11, 12)
+        x = xp.asarray(x)
         zpk = butter(3, 0.125, output='zpk')
-        y0 = self.filtfilt(zpk, x, padlen=0, axis=0)
-        y1 = self.filtfilt(zpk, np.swapaxes(x, 0, 1), padlen=0, axis=1)
-        xp_assert_equal(y0, np.swapaxes(y1, 0, 1))
-        y2 = self.filtfilt(zpk, np.swapaxes(x, 0, 2), padlen=0, axis=2)
-        xp_assert_equal(y0, np.swapaxes(y2, 0, 2))
+        y0 = self.filtfilt(zpk, x, padlen=0, axis=0, xp=xp)
+        y1 = self.filtfilt(
+            zpk, xp.asarray(np.swapaxes(x, 0, 1)), padlen=0, axis=1, xp=xp
```

Comment:
Would this be equivalent?
```suggestion
            zpk, xp.moveaxis(x, [0, 1], [1, 0]), padlen=0, axis=1, xp=xp
```

Reply:
I probably would. I'd still keep the test as is, for clarity. This is one of (rare) cases where swapaxes is more readable than moveaxis IMO.